main.tf

```
@@ -12,6 +12,11 @@ locals {
   ])
 }
 
+data "github_repository" "default" {
+  count = var.create_repository ? 0 : 1
+  name  = var.name
```

Or:
```suggestion
  name  = var.create_repository ? github_branch_protection.default.name : var.name
```

And then you can omit the count here and use the data source ref in the ID loc